server/memphis_handlers_poison_messages.go

```diff
@@ -118,35 +120,34 @@ func (s *Server) HandleNewMessage(msg []byte) {
 		PoisoningTime:   time.Now(),
 		DeliveriesCount: int(deliveriesCount),
 	}
-	filter := bson.M{
-		"station_name":      stationName.Ext(),
-		"message_seq":       int(messageSeq),
-		"producer.name":     producedByHeader,
-		"message.time_sent": poisonMessageContent.Time,
-	}
-	var newID = primitive.NewObjectID()
-	update := bson.M{
-		"$setOnInsert": bson.M{"_id": newID},
-		"$push":        bson.M{"poisoned_cgs": poisonedCg},
-		"$set":         bson.M{"message": messagePayload, "producer": producer, "creation_date": time.Now()},
+
+	id := GetPoisonMsgId(int(messageSeq), producedByHeader, poisonMessageContent.Time)
```

Comment:
id must contain the station name as well (all parameters appear in the filter section in the old implementation)